debian/rules

```diff
@@ -1,19 +1,42 @@
 #!/usr/bin/make -f
 
-include /usr/share/cdbs/1/rules/debhelper.mk
-include /usr/share/cdbs/1/class/autotools.mk
-include /usr/share/cdbs/1/rules/utils.mk
-include /usr/share/cdbs/1/class/gnome.mk
--include /usr/share/gnome-pkg-tools/1/rules/gnome-get-source.mk
-
-$(patsubst %,binary-predeb/%,$(DEB_ALL_PACKAGES)) ::
-	dh_girepository -p$(cdbs_curpkg) -l src:src/gvc -p /usr/lib/mutter \
-		/usr/lib/eos-shell
-	find debian/$(cdbs_curpkg) -name '*.la' -exec rm -f '{}' ';'
+include /usr/share/gnome-pkg-tools/1/rules/gnome-get-source.mk
+
+%:
+	dh $@ --with gir
+
+override_dh_autoreconf:
+	dh_autoreconf --as-needed
+
+ifeq ($(DEB_HOST_ARCH_OS),linux)
+CONFFLAGS = \
+	--enable-networkmanager \
+	--enable-systemd
+endif
 
-DEB_CONFIGURE_EXTRA_FLAGS += --enable-compile-warnings=yes
+override_dh_auto_configure:
+	dh_auto_configure -- \
+		--libdir=/usr/lib \
+		--libexecdir="\$${libdir}/eos-shell" \
+		--enable-compile-warnings \
+		$(CONFFLAGS)
 
-DEB_DH_MAKESHLIBS_ARGS_eos-shell = -Xlibeos-shell
+override_dh_install:
+	find debian/tmp -name '*.la' -delete
+	dh_install --list-missing
+
+override_dh_makeshlibs:
+	dh_makeshlibs -Xlibeos-shell
+
+override_dh_shlibdeps:
+	# eos-shell uses mutters private mutter-clutter-1.0.so etc.
+	dh_shlibdeps -l"usr/lib/$(DEB_HOST_MULTIARCH)/mutter"
+
+override_dh_girepository:
+	dh_girepository \
+		-l src:src/gvc \
+		-p /usr/lib/$(DEB_HOST_MULTIARCH)/mutter \
+		/usr/lib/eos-shell
 
-# List any files which are not installed
-common-binary-post-install-arch:: list-missing
+override_dh_strip:
+	dh_strip --dbgsym-migration='eos-shell-dbg (<< 3.20.2-2~)'
```

Comment:
We're not bumping the version of `eos-shell` to 3.20 as part of this though, right?